launch/launch/actions/execute_process.py

```diff
@@ -352,6 +352,26 @@ def parse(
                     )
                 kwargs['respawn_delay'] = respawn_delay
 
+        if 'sigkill_timeout' not in ignore:
+            sigkill_timeout = entity.get_attr('sigkill_timeout', data_type=float, optional=True)
+            if sigkill_timeout is not None:
+                if sigkill_timeout < 0.0:
+                    raise ValueError(
+                        'Attribute sigkill_timeout of Entity node expected to be '
+                        'a non-negative value but got `{}`'.format(sigkill_timeout)
+                    )
+                kwargs['sigkill_timeout'] = parser.parse_substitution(str(sigkill_timeout))
```

Comment:
Why does this need `parse_substitution`?

Reply:
Hmm if I just use ``kwargs['signkill_timeout'] = sigkill_timeout`` here, the tests fail with : 

```

    def normalize_to_list_of_substitutions(subs: SomeSubstitutionsType) -> List[Substitution]:
        """Return a list of Substitutions given a variety of starting inputs."""
        # Avoid recursive import
        from ..substitutions import TextSubstitution
    
        def normalize(x):
            if isinstance(x, Substitution):
                return x
            if isinstance(x, str):
                return TextSubstitution(text=x)
            raise TypeError(
                "Failed to normalize given item of type '{}', when only "
                "'str' or 'launch.Substitution' were expected.".format(type(x)))
    
        if isinstance(subs, str):
            return [TextSubstitution(text=subs)]
        if is_a_subclass(subs, Substitution):
            return [cast(Substitution, subs)]
>       return [normalize(y) for y in cast(Iterable, subs)]
E       TypeError: 'float' object is not iterable

```

Reply:
Probably because the ``ExecuteLocal`` class is expecting a ``SubstitutionsType`` for ``sigkill_timeout`` : https://github.com/ros2/launch/blob/edf68913560ebb127441aca513519b4f3a2559bd/launch/launch/actions/execute_local.py#L86-L89

Reply:
Yeah, but you don't need to use `parse_substitution()` though, only you need the `str()` conversion.

```suggestion
                kwargs['sigkill_timeout'] = str(sigkill_timeout)
```

Reply:
Fixed : https://github.com/ros2/launch/pull/667/commits/d45ea31bd89dbf4b12660384eaedbf062a95e9f2